feat: :sparkles: add data collection section (#122)
onlimit-study/dmp: sections/collection.qmd

# Data collection

Data for the ON LiMiT study will be collected from multiple sources,
including participants, healthcare professionals, lab technicians, and
electronic devices.

## Sources and methods

Participants will provide data through a variety of sources, the primary
being the Liva Healthcare app which will record lifestyle information
such as meals and physical activity, and questionnaires/site visit forms

> in REDCap. In addition connected devices (smart watch, smart scales,

Reviewer: ```suggestion
in REDCap. Connected devices such as smart watch, smart scales,
```

MyFood24) on a monthly basis.

Sensor and scanner data (SENS, Libre, Fibro scans, and DXA scans) will
be uploaded at clinic visits, as we expect to sign off on a visit as
soon as it is finished.

<!--TODO check the agreement with Liva once signed-->

## Data formats

REDCap: Structured tabular data in CSV imported to GenomeDK via API.\
Liva app: CSV exports via FTP to GenomeDK.\
- Watch, scales: Direct import into Liva app.\
MyFood24: CSV import from secure website into GenomeDK.\
Scanners: CSV files uploaded into REDCap or directly to GenomeDK.

## Estimated raw data volumes for the feasibility study

REDCap clinical data: Unknown.\
Liva app data: Unknown.\
MyFood24 dietary logs: Unknown.\
Sensor data (SENS, Libre): Unknown.\
Scanner data:\
- VO2Max: 20KB times 1 reading per visit, 4 visits = 20x4x24 = 2MB.\
- Finapres: 60KB times 8 readings per visit, 6 visits = 60x8x6x24 =
69MB.

## Data aggregation and storage

The REDCap database serves as the central hub for most clinical and
scanner data, including manual uploads from devices such as SENS and
Libre, and potentially iMotions. The Liva app aggregates participant

> lifestyle data and connected device readings. In addition there will be

Reviewer: ```suggestion
lifestyle data and connected device readings. There will also be
```

test results (like VO2Max and Finapres) which will be sent to GenomeDK
directly, the same applies to data from MyFood24.

GenomeDK's HPC facility will serve as the final repository for processed
and collated data, ensuring secure storage and computational capacity
for analysis.

See the [Storage, backup, recovery, and access](storage.qmd) for further
details.

## Quality assurance

Validation checks: Automated checks in REDCap to ensure completeness and
consistency of clinical data (Responsible: Data Architect).

Device data verification: Uploaded sensor data will be checked against
expected ranges and timestamps (Responsible: Site Staff using REDCap).

Monitoring for missing/anomalous data: Regular reports will identify

> gaps or anomalies, with follow-up procedures for correction and will be

Reviewer: ```suggestion
any anomalies and report on the results (if any). We will do any follow-up procedures to correct for these issues, which we will
```

run during transfer to GenomeDK. (Responsible: Data Architect).

<!--TODO Do we need any checks on the API and FTP transfers?-->

## Visualisation of data flow in ON LiMiT

Underneath the diagram is a more detailed description of the components
and their interactions.

```{mermaid}
%%| label: fig-c4-context
%%| fig-cap: "Data flow in the ON LiMiT study, who generates data, where is does it enter, and how is it transmitted to storage."
%%| column: page-right
flowchart LR

> Lab(["<b>Lab staff</b><br> collecting and analysing biological samples from participants."])

Reviewer: ```suggestion
Lab(["<b>Lab staff</b><br>collecting and analysing biological samples from participants."])
```
